src/smartpeak/source/core/SequenceSegmentProcessors/CalculateCalibration.cpp

```diff
@@ -58,33 +71,82 @@ namespace SmartPeak
     LOGD << "START optimizeCalibrationCurves";
     getFilenames(filenames_I);
 
-    std::vector<size_t> standards_indices;
+    // Complete user parameters with schema
+    ParameterSet params(params_I);
+    params.merge(getParameterSchema());
+
+    std::vector<std::tuple<std::string, std::string>> excluded_points;
+    auto excluded_points_param = params.findParameter("CalculateCalibration", "excluded_points");
+    if ((!excluded_points_param) || (excluded_points_param->getType() != "string_list"))
+    {
+      throw std::invalid_argument("CalculateCalibration::excluded_points not found or wrong type");
+    }
+    CastValue excluded_points_value;
+    Utilities::parseString(excluded_points_param->getValueAsString(), excluded_points_value);
+    for (const std::string excluded_point_value : excluded_points_value.sl_)
+    {
+      auto sep_pos = excluded_point_value.find(";");
+      if (sep_pos != std::string::npos)
+      {
+        auto sample_name = excluded_point_value.substr(0, sep_pos);
+        auto component_name = excluded_point_value.substr(sep_pos+1, excluded_point_value.size()-sep_pos);
+        excluded_points.push_back(std::make_tuple(sample_name, component_name));
+      }
+    }
+      
     // get all standards
+    std::vector<size_t> all_standards_indices;
     this->getSampleIndicesBySampleType(
       sequenceSegmentHandler_IO,
       sequenceHandler_I,
       SampleType::Standard,
-      standards_indices
+      all_standards_indices
     );
 
     // check if there are any standards to calculate the calibrators from
-    if (standards_indices.empty()) {
+    if (all_standards_indices.empty()) {
       throw std::invalid_argument("standards_indices argument is empty.");
     }
 
-    std::vector<OpenMS::FeatureMap> standards_featureMaps;
-    for (const size_t index : standards_indices) {
-      standards_featureMaps.push_back(sequenceHandler_I.getSequence().at(index).getRawData().getFeatureMap());
-    }
-
     // add in the method parameters
     OpenMS::AbsoluteQuantitation absoluteQuantitation;
     Utilities::setUserParameters(absoluteQuantitation, params_I);
 
     absoluteQuantitation.setQuantMethods(sequenceSegmentHandler_IO.getQuantitationMethods());
     std::map<std::string, std::vector<OpenMS::AbsoluteQuantitationStandards::featureConcentration>> components_to_concentrations;
     std::map<std::string, std::vector<OpenMS::AbsoluteQuantitationStandards::featureConcentration>> outlier_components_to_concentrations;
-    for (const OpenMS::AbsoluteQuantitationMethod& row : sequenceSegmentHandler_IO.getQuantitationMethods()) {
+    std::map<std::string, std::vector<OpenMS::AbsoluteQuantitationStandards::featureConcentration>> excluded_components_to_concentrations;
```

Comment:
Rebuilding these structures at each call (along with their modifications below) will be quite resource intensive.  I would recommend creating a separate method in the `SequenceSegmentHandler` to build the `features_concentrations_pruned` and remove all 0 values as done on lines 89 to 120 in the original code which can be used by both `OptimizeCalibration` and `CalculateCalibration` as needed.